static/app/components/profiling/flamegraph.tsx

```diff
@@ -35,41 +35,41 @@ interface FlamegraphProps {
 function Flamegraph(props: FlamegraphProps): ReactElement {
   const flamegraphTheme = useFlamegraphTheme();
   const [{sorting, view, xAxis}, dispatch] = useFlamegraphPreferences();
-  const [{activeProfileIndex}, dispatchActiveProfileIndex] = useFlamegraphProfiles();
+  const [{threadId}, dispatchThreadId] = useFlamegraphProfiles();
 
   const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
 
   const flamegraph = useMemo(() => {
-    if (
-      !props.profiles.profiles[activeProfileIndex ?? props.profiles.activeProfileIndex]
-    ) {
-      // This could happen if activeProfileIndex was initialized from query string, but for some
-      // reason the profile was removed from the list of profiles.
+    if (typeof threadId !== 'number') {
       return FlamegraphModel.Empty();
     }
+
+    // This could happen if activeProfileIndex was initialized from query string, but for some
+    // reason the profile was removed from the list of profiles.
```

Comment:
```suggestion
    // This could happen if threadId was initialized from query string, but for some
    // reason the profile was removed from the list of profiles.
```